src/statement-list/statement_list_parse_tree.h

```diff
@@ -85,106 +75,108 @@ class statement_list_parse_treet
+    /// Version of the module.
     float version;
 
-    // Variable declarations
+    /// Input variable declarations.
     var_declarationst var_input;
+    /// Inout variable declarations.
     var_declarationst var_inout;
+    /// Output variable declarations.
     var_declarationst var_output;
+    /// Temp variable declarations.
+    var_declarationst var_temp;
+    /// Constant variable declarations.
+    var_declarationst var_constant;
 
-    // Body
+    /// List of all networks of this module.
     networkst networks;
 
-    /// Create the function \p name with a specific \p version.
-    explicit functiont(const irep_idt &name, const float version)
-      : name(name), version(version)
-    {
-    }
+    /// Adds a variable declaration to the list of input variables.
+    /// \param declaration: Variable declaration to be added.
+    void add_var_input_entry(var_declarationt &declaration);
+    /// Adds a variable declaration to the list of inout variables.
+    /// \param declaration: Variable declaration to be added.
+    void add_var_inout_entry(var_declarationt &declaration);
+    /// Adds a variable declaration to the list of output variables.
+    /// \param declaration: Variable declaration to be added.
+    void add_var_output_entry(var_declarationt &declaration);
+    /// Adds a variable declaration to the list of temp variables.
+    /// \param declaration: Variable declaration to be added.
+    void add_var_temp_entry(var_declarationt &declaration);
+    /// Adds a variable declaration to the list of constant variables.
+    /// \param declaration: Variable declaration to be added.
+    void add_var_constant_entry(var_declarationt &declaration);
+    /// Adds a network to the function.
+    /// \param network: Network to be added.
+    void add_network(networkt &network);
+
+    // Disallow copy construction and copy assignment, but allow move
+    // construction and move assignment.
+    tia_modulet(const tia_modulet &) = delete;
+    tia_modulet &operator=(const tia_modulet &) = delete;
+    tia_modulet(tia_modulet &&) = default;
+    tia_modulet &operator=(tia_modulet &&) = default;
+
+    /// Create the module \p name with a specific \p version.
+    /// \param name: Name of the module.
+    /// \param version: Version of the module.
+    explicit tia_modulet(const irep_idt &name, const float version);
+  };
+
+  /// Structure for a simple function in Statement List. Includes fields for
+  /// its name, version, variable declarations and networks.
+  struct functiont : tia_modulet
+  {
+    /// FC-exclusive return value.
+    exprt return_value;
+
+    /// Create the function \p name with a specific \p version and a
+    /// \p return_value.
+    /// \param name: Name of the function.
+    /// \param version: Version of the function.
+    /// \param return_value: Type of the function's return value.
+    explicit functiont(
+      const irep_idt &name,
+      const float version,
+      const exprt &return_value);
```

Comment:
Comment says `type`, name says `value`. If it is a type, probably best to change this to a `typet` and call it `return_type`.